src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticConverter.cs

```diff
@@ -24,7 +25,18 @@ public static Diagnostic Convert(RazorDiagnostic razorDiagnostic, SourceText sou
             throw new ArgumentNullException(nameof(sourceText));
         }
 
-        var diagnostic = new Diagnostic()
+        var projects = documentSnapshot is null
+            ? Array.Empty<VSDiagnosticProjectInformation>()
+            : [
+                new VSDiagnosticProjectInformation()
+                {
+                    Context = string.Empty,
+                    ProjectIdentifier = documentSnapshot.Project.TryGetProjectContext(documentSnapshot.FilePath, out var context) ? context.Id : string.Empty,
```

Comment:
This seems overkill. Isn't it just `documentSnapshot.Project.Key.Id`?

Reply:
🤦